covid19_sfbayarea/data/sonoma.py

```diff
@@ -0,0 +1,261 @@
+        type = transmission_type_conversion[type]
+        transmissions[type] = parse_int(number)
+    return transmissions
+
+def transform_tests(tests_tag: element.Tag) -> Dict[str, int]:
+    tests = {}
+    rows = get_rows(tests_tag)
+    for row in rows:
+        result, number, *rest = get_cells(row)
+        lower_res = result.lower()
+        tests[lower_res] = parse_int(number)
+    return tests;
+
+def transform_gender(tag: element.Tag) -> Dict[str, int]:
+    """
+    Transform function for the cases by gender table.
+    Takes in a BeautifulSoup tag for a table and returns a dictionary
+    in which the keys are strings and the values integers
+    """
+    categories = {}
+    rows = get_rows(tag)
+    gender_string_conversions = {'Males': 'male', 'Females': 'female'}
+    for row in rows:
+        gender, cases, *rest = get_cells(row)
+        if gender not in gender_string_conversions:
+            raise FormatError('An unrecognized gender has been added to the gender table')
+        categories[gender_string_conversions[gender]] = parse_int(cases)
+    return categories
+
+def transform_age(tag: element.Tag) -> List[Dict[str, Union[str, int]]]:
+    """
+    Transform function for the cases by age group table.
+    Takes in a BeautifulSoup tag for a table and returns a list of
+    dictionaries in which the keys are strings and the values integers
+    """
+    categories: List[Dict[str, Union[str, int]]] = []
+    rows = get_rows(tag)
+    for row in rows:
+        group, cases, *rest = get_cells(row)
+        raw_cases = parse_int(cases)
+
+        element: Dict[str, Union[str, int]] = {'group': group, 'raw_cases': raw_cases}
+        categories.append(element)
+    return categories
+
+def get_unknown_race(race_eth_tag: element.Tag) -> int:
+    """
+    Gets the notes under the 'Cases by race and ethnicity' table to find the
+    number of cases where the person's race is unknown
+    """
+    parent = race_eth_tag.parent
+    note = parent.find('p').text
+    matches = re.search(r'(\d+) \(\d{1,3}%\) missing race/ethnicity', note)
+    if not matches:
+        raise FormatError('The format of the note with unknown race data has changed')
+    return(parse_int(matches.groups()[0]))
+
+def transform_race_eth(race_eth_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in the BeautifulSoup tag for the cases by race/ethnicity table and
+    transforms it into an object of form:
+    'race_eth': {'Asian': -1, 'Latinx_or_Hispanic': -1, 'Other': -1, 'White':-1, 'Unknown': -1}
+    NB: These are the only races reported seperatley by Sonoma county at this time
+    """
+    race_cases = {
+        'African_Amer': 0,
+        'Asian': 0,
+        'Latinx_or_Hispanic': 0,
+        'Native_Amer':0,
+        'Multiple_Race':0,
+        'Other': 0,
+        'Pacific_Islander': 0,
+        'White': 0,
+        'Unknown': 0
```

Comment:
Seems like the base value for `African_Amer`, `Native_Amer`, `Multiple_Race`, and `Pacific_Islander` should probably be either `-1` or not present here to clarify that the counts are unknown rather than 0 (since they’re included in `Asian` or `Other`).